### PR TITLE
fix(amplify-category-function): check for new function when adding permissions

### LINKED ISSUE #7970: Creating Function with access to another services
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.15.1

### Amplify CLI Version

5.3.0

### What operating system are you using?

Mac

### Amplify Categories

storage, function, api

### Amplify Commands

add

### Describe the bug

I'm trying to create a new function with access to another services as Storage, API or Another functions, so when I tried to give the access through the console I got the error.
image

### Expected behavior

I expected that function must have access to the another services.

### Reproduction steps

image

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by fossamagna**
#### Description of changes

A function attempting to add permissions is handled as new function when is not found in amplify meta.

#### Issue #, if available

fix #7970

#### Description of how you validated changes

- Unit Testing `yarn lerna --scope amplify-category-function run test`
- Manual Testing

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-function/src/__tests__/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.test.ts [hunk 1]
```diff
@@ -103,3 +128,127 @@ test('env resources for CFN for auth and storage for api', async () => {
...
+    };
+    const answers = await askExecRolePermissionsQuestions((contextStub as unknown) as $TSContext, 'lambda3', undefined);
+    expect(answers).toMatchSnapshot();
+  });
+
+  it('returns permissions for exists function', async () => {
+    const inquirerMock = inquirer as jest.Mocked<typeof inquirer>;
+
+    inquirerMock.prompt.mockResolvedValueOnce({ categories: ['function'] });
+    //inquirerMock.prompt.mockResolvedValueOnce({ resources: ['lambda2'] });
```
Comment (jhockett) [inline packages/amplify-category-function/src/__tests__/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.test.ts:215]: nit: remove this line if no longer needed

File: packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.ts [hunk 2]
```diff
@@ -80,12 +80,14 @@ export const askExecRolePermissionsQuestions = async (
       // A Lambda function cannot depend on itself
       // Lambda layer dependencies are handled seperately, also apply the filter if the selected resource is within the function category
       // but serviceName argument was no passed in
-      const selectedResource = _.get(amplifyMeta, [categoryName, resourceNameToUpdate]);
-
       if (serviceName === ServiceName.LambdaFunction || selectedCategory === categoryName) {
+        const selectedResource = _.get(amplifyMeta, [categoryName, resourceNameToUpdate]);
+        // A new function resource is not exsits in amplifyMeta yet
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.ts:85]:
```suggestion
        // A new function resource does not exist in amplifyMeta yet
```

**Review by jhockett — APPROVED** [review]
LGTM, thanks for the contribution @fossamagna! I left two minor comments

File: packages/amplify-category-function/src/__tests__/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.test.ts (hunk 1 above)
Comment (fossamagna) [reply packages/amplify-category-function/src/__tests__/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.test.ts]: Thank you. I remove this line.

**Pull request merged**